src/main/java/traincraft/blocks/assemblytables/TileAssemblyTable.java

```diff
@@ -0,0 +1,202 @@
+    }
+    
+    /**
+     * This function handles recipe checking. Will set the output item if recipe works.
+     */
+    public void onInventoryChanged() {
+        //for loop to compare between ingredients, similar to workbench
+        if(this.world == null || !this.world.isRemote){
+            this.markDirty();
+            //clear output inventory area
+            outputInventory.clear();
+        
+            if(!craftingInventory.isEmpty()){
+                //filter through the train recipes and find the first match (there should only be one match, but just in case duplicate recipes or sth.
+                AssemblyTableRecipe.ASSEMBLY_TABLE_RECIPES.stream().filter(recipe -> recipe.betterMatches(craftingInventory)).findFirst().ifPresent(recipe -> {
+                    if (this.tier == recipe.getTier()){ //make sure correct tier before doing.
+                        recipeInUse = recipe;
+                        outputInventory.setInventorySlotContents(0, recipe.getRecipeOutput().copy());
+                        containerAssemblyTable.detectAndSendChanges();
+                        this.syncToClient();
+                    }
+                });
+            }
+    
+            //update things that need update
+            containerAssemblyTable.detectAndSendChanges();
+        }
+    }
+    
+    //used break rather than harvest because want to drop items even if wrong tool or other method used to break. Like chests.
+    @Override
+    public void onBlockBreak(IBlockState state){
+        //drop items from crafting inventory
+        for(int i = 0; i < craftingInventory.getSlots(); ++i){
+            if (craftingInventory.getStackInSlot(i) != ItemStack.EMPTY){
+                this.world.spawnEntity(new EntityItem(world, pos.getX(), pos.getY(), pos.getZ(), craftingInventory.getStackInSlot(i)));
+            }
+        }
+        
+        //drop items from storage inventory
+        for(int i = 0; i < storageInventory.getSlots(); ++i){
+            if (storageInventory.getStackInSlot(i) != ItemStack.EMPTY){
+                this.world.spawnEntity(new EntityItem(world, pos.getX(), pos.getY(), pos.getZ(), storageInventory.getStackInSlot(i)));
+            }
+        }
+    }
+    
+    @Override
+    public boolean hasCapability(@Nonnull Capability<?> capability, @Nullable EnumFacing facing){
+        if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY) {
+            return true;
+        }
+        return super.hasCapability(capability, facing);
+    }
+    
+    @Nullable
+    @Override
+    public <T> T getCapability(@Nonnull Capability<T> capability, @Nullable EnumFacing facing){
+        if(capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY){
+            IItemHandler itemHandler = this.getStorageInventory();
+            if(itemHandler != null){
+                return (T) itemHandler;
+            }
+        }
+        return super.getCapability(capability, facing);
+    }
```

Comment:
You can just override "public IItemHandler getInventory(@Nullable EnumFacing side)" (example in the TileDitillery class). This than handles the capabilities

Reply:
This does not handle the reading and writing nbt because that looks for the IInventory, which I don't use, so I just implemented writeNBT and readNBT, both short functions, and it'll be more efficient too. Removed the capability functions from Assembly Table TileEntity